txpipe/test/test_utils.py

```diff
@@ -0,0 +1,34 @@
+from ..utils.misc import unique_list, hex_escape, multi_where
+import numpy as np
+
+def test_escape():
+    assert hex_escape(chr(1))=='\\x01'
+    assert hex_escape(chr(7) + chr(11) + chr(12))=='\\x07\\x0b\\x0c'
+
+    assert hex_escape("aaa\nbbb") == "aaa\nbbb"
+    assert hex_escape("xxx\nyyy", replace_newlines=True) == "xxx\\x0ayyy"
+
+def test_unique_list():
+
+    x = [1, 1, 1, 1]
+    assert unique_list(x) == [1]
+
+    x = [1, 1, 2, 1, 1]
+    assert unique_list(x) == [1, 2]
+
+    x = [1, 1, 2, 1, 1, 2, 3, 3]
+    assert unique_list(x) == [1, 2, 3]
+
+    x = [-1, "cat", -1, "cat", "dog"]
+    assert unique_list(x) == [-1, "cat", "dog"]
+
+def test_multi_where():
+    x = np.random.randint(0, 10, 100)
+    w = [1, 2, 3]
+
+    a = multi_where(x, w)
+
+    for m in w:
+        assert np.array_equal(np.where(x==m)[0], a[m])
+
+    assert a.keys() == set(w)
```

Comment:
What is the use case where this is faster than numpy?

I tested this unit test and timed `multi_where` vs `[np.where(x==m)[0] for m in w]` an found:
```
multi_where time =  5.602836608886719e-05
numpy where time =  2.1219253540039062e-05
```
So I thought, ok, it's just too small for the multi_where advantage to kick in.  Let's try bigger w and x:
```
x = np.random.randint(0, 100, 10000)
w = np.unique(np.random.randint(0,100,20))

t0 = time.time()
a = multi_where(x, w)
t1 = time.time()
b = {m: np.where(x==m)[0] for m in w}
t2 = time.time()

print('multi_where time = ',t1-t0)
print('numpy where time = ',t2-t1)
```
This yielded
```
multi_where time =  0.0028181076049804688
numpy where time =  0.0002880096435546875
```
So an even bigger advantage to numpy.  I'm sure you have something in mind, but I would guess that the (C-implemented) numpy function will usually win out over python loops.

Reply:
The original use case was when the matches are heterogeneous, in particular because I was calling some tomographic bins 0, 1, 2, 3 and "all" (for the 2D case).  But you're right that it probably isn't worth the slow down, and in the end I was only using this in one place.  So I've removed it. Good point - thanks!